### PR TITLE
Feature/sessions

### PR DESCRIPTION
**Pull request by ccali11**
@shanejearley - PR for #255. Not much to note at this point except that it is working well on first connect (signUp / login), adding a sub-account to a user's primary account, and logging out. Will continue maturing and protecting more routes as we build out the users backend and API. Let me know if you have any questions.

### CODE REVIEW

File: apps/web/src/composables/users.ts [hunk 1]
```diff
@@ -16,6 +16,19 @@ const user = ref<User | null>(null)
 export default function useUsers () {
     const { ssvManager } = useSSV()
 
+    async function getUserFromAPI() {
```
Comment (shanejearley) [inline apps/web/src/composables/users.ts:19]: `getUser`

File: apps/web/src/composables/users.ts [hunk 2]
```diff
@@ -16,6 +16,19 @@ const user = ref<User | null>(null)
...
+        const requestOptions = {
+            method: 'GET',
+            headers: { 
+                'Content-Type': 'application/json'
+            }
+        }
+        const response = await fetch(`${usersBaseURL}/users`, requestOptions)
```
Comment (shanejearley) [inline apps/web/src/composables/users.ts:26]: `${usersBaseURL}/user`

File: apps/web/src/composables/users.ts [hunk 3]
```diff
@@ -80,6 +92,8 @@ export default function useUsers () {
             }
             const response = await fetch(`${usersBaseURL}/users/add-sub-account`, requestOptions)
             const json = await response.json()
```
Comment (shanejearley) [inline apps/web/src/composables/users.ts:94]: const { user } = await response.json()

File: apps/web/src/sessions.config.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import Session from 'supertokens-web-js/recipe/session'
+import useEnvironment from '@/composables/environment'
+
+const { usersBaseURL } = useEnvironment()
+
+export const SuperTokensWebJSConfig = {
+  appInfo: {
+      apiDomain: 'http://localhost:4000', // Replace with usersBaseURL
```
Comment (shanejearley) [inline apps/web/src/sessions.config.ts:8]: Yes replace.

File: services/users/src/index.ts [hunk 5]
```diff
@@ -1,20 +1,42 @@
...
+        methods: ['GET', 'PUT', 'POST', 'DELETE'],
+        credentials: true
+    })
+)
+
+/** Expose all the APIs from SuperTokens to the client */
+app.use(middleware())
+
 app.use('/auth', auth)
+app.use('/signupLogin', signupLogin)
```
Comment (shanejearley) [inline services/users/src/index.ts:33]: Just use /auth to do this.

File: services/users/src/routes/signupLogin.ts [hunk 6]
```diff
@@ -15,7 +16,7 @@ router.use('/', async (req: express.Request, res: express.Response) => {
     const { provider, address, currency, message, signedMessage } = body as SignupLoginCredentials
     const user = userCollection.find(user => user.address === address.toLowerCase())
     if (user && !user?.accounts) {  // signup
-        console.log('got into this block in signupLogin.ts')
+        console.log('-----------------got into this block in signupLogin.ts-----------------')
```
Comment (shanejearley) [inline services/users/src/routes/signupLogin.ts:19]: Clean this all up.

File: services/users/src/routes/users.ts [hunk 7]
```diff
@@ -1,7 +1,30 @@
...
+import { verifySession } from 'supertokens-node/recipe/session/framework/express'
+import { SessionRequest } from 'supertokens-node/framework/express'
 const router = express.Router()
 
+router.get('/', verifySession(), (req: SessionRequest, res: express.Response) => {
+    const address = req.session?.getUserId()
+    const user = userCollection.find(user => user.address === address?.toLowerCase())
+    res.setHeader('Content-Type', 'application/json')
+    res.status(200)
+    if (user) {
```
Comment (shanejearley) [inline services/users/src/routes/users.ts:12]: Just return { user } or { error }.

**Review by shanejearley — APPROVED** [review]

File: apps/web/src/composables/users.ts (hunk 3 above)
Comment (ccali11) [reply apps/web/src/composables/users.ts:94]: Can't call it `user` due to naming collision, so called it `userData` instead.

File: services/users/src/routes/users.ts (hunk 7 above)
Comment (ccali11) [reply services/users/src/routes/users.ts:12]: Not sure what you mean by this one. I simplified it I guess, but not sure it's what you want.

**Review by shanejearley — APPROVED** [review]
The updates look good, it'll be easier to explain one or two of them next time we meet, and we can discuss more changes then. Fully approved to merge.

**Comment by ccali11** [comment]
@ccali11 - Change `signupLogin` to `login`, you doofus.

**Pull request merged**